src/vs/workbench/services/editor/browser/editorOverrideService.ts

```diff
@@ -91,11 +91,8 @@ export class EditorOverrideService extends Disposable implements IEditorOverride
 		}
 
 		// Always ensure inputs have populated resource fields
-		if (editor instanceof DiffEditorInput) {
-			if ((!editor.modifiedInput.resource || !editor.originalInput.resource)) {
-				return { editor, options, group };
-			}
-		} else if (!editor.resource) {
+		const resource = EditorResourceAccessor.getCanonicalUri(editor, { supportSideBySide: SideBySideEditor.PRIMARY });
```

Comment:
`EditorResourceAccessor` is a utility specifically for obtaining the `resource` from an editor including support for side by side editors. The `SideBySideEditor.PRIMARY` will be the `modifiedInput` for a diff editor.

I believe the check for `editor.originalInput.resource` was not needed for the overrides to work (it is gone with my change).